### PR TITLE
(128) Activity form uses govuk form builder

### PR DESCRIPTION
**Pull request by lozette**
## Changes in this PR

Trello: https://trello.com/c/AvIykhu2/128-refactor-activity-form-to-use-new-form-builder

All forms in the application now use `govuk_design_system_formbuilder`. 

## Next steps

- [ ] Is an ADR required? An ADR should be added if this PR introduces a change to the architecture.
- [x] Is a changelog entry required? An entry should always be made in `CHANGELOG.md`, unless this PR is a small tweak which has to impact outside the development team.
- [ ] Do any environment variables need amending or adding?

### CODE REVIEW

File: app/models/activity.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ class Activity < ApplicationRecord
   validates_presence_of :identifier
   validates_uniqueness_of :identifier
 
+  attribute :recipient_region, :string, default: "998"
```
Comment (tahb) [inline app/models/activity.rb]: I think these defaults will work fine for fund activities but I wonder how we'll be able to support these being changed for programmes and projects, any ideas?

File: app/helpers/codelist_helper.rb [hunk 2]
```diff
@@ -1,15 +1,6 @@
 # frozen_string_literal: true
 
 module CodelistHelper
-  # TODO: Remove `yaml_to_options` when all forms are migrated from simple form to the govuk form builder,
```
Comment (tahb) [inline app/helpers/codelist_helper.rb]: Good to see us closing the loop and binning the old version 👍

**Comment by lozette** [comment]
I think the asterisk for a required field is not part of the govuk design standard: https://govuk-elements.herokuapp.com/form-elements/#form-optional-fields

File: app/models/activity.rb (hunk 1 above)
Comment (lozette) [reply app/models/activity.rb]: Possibly a set of env variables? 

Or maybe a helper along the lines of ActivityHelper. Although if the values needed to change than env variables are probably more flexible?

**Comment by tahb** [comment]
@lozette Ah brilliant 👌 Really what we'd need if anything would be to add `(optional)` in everywhere else however [this card should add validation to everything](https://trello.com/c/GiHeA5QU).

File: app/models/activity.rb (hunk 1 above)
Comment (tahb) [reply app/models/activity.rb]: I hadn't thought of environment variables, how might that look?

I wonder if having some initializer helpers might help. Something like:

```
class Activity

def setup_hierarchy_defaults
  case hierachy.class.name
  when fund setup_fund_defaults
…
end

def setup_fund_defaults
  self.recipient_region = "998"
  …
end

def setup_programme_detaults
  self.recipient_region = "2"
  …
end
```

```
class Staff::ActivitiesController < Staff::BaseController
  include ActivityHelper
  def create 
    @activity.hierarchy = hierarchy
    @activity.set_up_hierachy_defaults
    @activity.save(validate: false)
…

Comment (lozette) [reply app/models/activity.rb]: I like the initialiser helpers more, env variables would be clunky and I don't think these values are going to change often (maybe?)

**Pull request merged**